src/trudp_channel.c

```diff
@@ -623,6 +631,29 @@ int trudpChannelProcessReceivedPacket(trudpChannelData *tcd, uint8_t *data,
         trudpSendQueueDelete(tcd->sendQueue, sqd);
         tcd->td->stat.sendQueue.size_current--;
 
+        if (tcd->td->channel_key == tcd->channel_key) {
+            teoMapIterator it;
+            teoMapElementData *el;
+            uint32_t min_timeout_sq = UINT32_MAX;
+            uint64_t current_time = teoGetTimestampFull();
+            teoMapIteratorReset(&it, tcd->td->map);
+            while((el = teoMapIteratorNext(&it))) {
+                trudpChannelData *tcd = (trudpChannelData *)teoMapIteratorElementData(el, NULL);
+                uint32_t timeout_sq = trudpChannelSendQueueGetTimeout(tcd, current_time);
+                if(timeout_sq < min_timeout_sq) {
+                  min_timeout_sq = timeout_sq;
+                  tcd->td->expected_max_time = timeout_sq;
+                  tcd->td->channel_key = tcd->channel_key;
+                }
+                if(!min_timeout_sq) break;
+            }
```

Comment:
Code like this exist in two places - here and in trudp.c:189, would much better to extract it to function